### PR TITLE
User can only have 2 initials in UI

### PR DESCRIPTION
**Pull request by voodoo**
Addresses:  "You should create a method that takes user.name.initials as its argument. We are already properly extracting initials, we just want fewer of them. So make a helper like you’ve done that takes in all initials and returns a max of two. If there are more than two, keep the first and last initial and drop any in the middle."

I think this kind of works - but I've not done this type of process in a while so I expect it is not perfect ... yet.

### CODE REVIEW

**Comment by voodoo** [comment]
I'm looking to make a 'good' PR. So go over with a fine tuned comb and shout. I already see a couple things I would like to have done better.

File: Gemfile.lock [hunk 1]
```diff
@@ -437,6 +437,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-23
+  arm64-darwin-24
```
Comment (krschacht) [inline Gemfile.lock:440]: I don’t know what this line is doing, do you? I’m not opposed to adding it if it’s helping something, but if it’s just a stray line we should remove it.

File: app/views/layouts/_user_avatar.erb [hunk 2]
```diff
@@ -11,5 +11,5 @@
         <%= classes %>
         "
 >
-  <%= user.name.initials&.upcase %>
+  <%= only_user_initials(user.name.initials).upcase %>
```
Comment (krschacht) [inline app/views/layouts/_user_avatar.erb:14]: The method name is a bit confusing. I find myself thinking, “but isn’t name.initialis” only the user initials? I don’t expect that to return something other than initials.

A better name might be: “only_two_initials”

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,4 +1,21 @@
 module ApplicationHelper
+
+  def only_user_initials(initials, limit: 2)
+    letters = initials.to_s.downcase.strip.split("")
```
Comment (krschacht) [inline app/helpers/application_helper.rb:4]: I suspect this whole method can be a lot simpler. Probably this will work:

def …
  return initials if initials.nil? || initials.length <= 2
  initials[0] + initials[-1]
end

I haven’t tested that so I may be overlooking something, but I think that’s a simple way to grab the first letter and the last letter.

File: app/views/layouts/_user_avatar.erb (hunk 2 above)
Comment (krschacht) [inline app/views/layouts/_user_avatar.erb:14]: Also, I think I’d keep the &.upcase rather than removing &, at least the method the way I wrote it returns a nil if it’s passed a nil, which seems like reasonable behavior

File: test/helpers/application_helper_test.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+require "test_helper"
+
+class ApplicationHelperTest < ActionView::TestCase
+  
+test "only at most 2 initials with default limit" do
+    assert_equal "jz", only_user_initials("jdz")
+    assert_equal "pl", only_user_initials("Plllll")
```
Comment (krschacht) [inline test/helpers/application_helper_test.rb:7]: It’s a bit unexpected that this method would change case. The way I wrote it preserves whatever case was passed in. I think that’s what most users would expect.

Also, this second test is bad because the fact that it returns “pl” does not confirm that it’s returning the first letter and the last letter. There are a lot of “l”s

File: test/helpers/application_helper_test.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  end    
+
+  test "can have numbers" do
+    assert_equal "p2", only_user_initials("Pvv2")
+  end  
+
+  test "can have spaces" do
+    assert_equal "pq", only_user_initials("P vv qQ")
+  end  
+
```
Comment (krschacht) [inline test/helpers/application_helper_test.rb:26]: It’s worth adding a test for the nil case

**Review by krschacht — CHANGES_REQUESTED** [review]
Alright, you asked for it so I gave it a full review. :)

File: Gemfile.lock (hunk 1 above)
Comment (voodoo) [reply Gemfile.lock:440]: That was added automatically. I just assumed cause I'm on a mac m1. I can take it out ... maybe :) I'll look into git-ing it out.

**Comment by voodoo** [comment]
Ok, I added those awesome/needed/helpful changes. I appreciate you doing that.

**Comment by voodoo** [comment]
DAMN! Did no mean to CLOSE the PR.

**Comment by voodoo** [comment]
These code versioning tools are awesome but I  feel like I'm running with scissors.  I might have done something wrong - but I learn. Really, make any other observations.

File: app/views/layouts/_user_avatar.erb (hunk 2 above)
Comment (voodoo) [reply app/views/layouts/_user_avatar.erb]: only_two_initials makes great sense

Comment (voodoo) [reply app/views/layouts/_user_avatar.erb]: &.upcase on the method.

File: test/helpers/application_helper_test.rb (hunk 4 above)
Comment (voodoo) [reply test/helpers/application_helper_test.rb]: No longer changing case. Took out a few l's.

File: app/views/layouts/_user_avatar.erb (hunk 2 above)
Comment (voodoo) [reply app/views/layouts/_user_avatar.erb]: Will do.

File: test/helpers/application_helper_test.rb (hunk 4 above)
Comment (voodoo) [reply test/helpers/application_helper_test.rb]: True, does not change case now.

**Comment by voodoo** [comment]
I appreciate your patience and your coding skills. You can delete. I'll do better on the next one.

File: Gemfile.lock (hunk 1 above)
Comment (voodoo) [reply Gemfile.lock]: I removed the arm64-darwin-24. I think lol. Thanks for your patience.

Comment (voodoo) [reply Gemfile.lock]: Running the test, it tries to stuff it back in there.

File: app/views/layouts/_user_avatar.erb (hunk 2 above)
Comment (voodoo) [reply app/views/layouts/_user_avatar.erb]: only_two_initials was an even better name but I went with at_most_two_initials to cover the case for one initial

**Comment by voodoo** [comment]
Hopefully you have not ghosted or unsubscribed or something.  I've not done a pull request in about a decade. That does not explain my carelessness - but I do care!

**Comment by krschacht** [comment]
Hey! A busy couple days at my day job, but I’ll definitely circle back soon. Sorry for the delay!

**Comment by voodoo** [comment]
No worries Keith  - I appreciate you letting me know.

File: Gemfile.lock (hunk 1 above)
Comment (krschacht) [reply Gemfile.lock]: Some quick googling suggests that you may be on a newer version of the Mac operating system and that’s why it’s added, but I didn’t dig into it much. It’s probably fine to leave it but it’s also fine to take it out.

**Review by krschacht — APPROVED** [review]
All your changes were good ones. I did some final cleanup and as soon as the test suite passes I’ll merge this in. Congrats on your first PR!

You can look at the commit log to see my final changes. Notable takeaways:

1) Your code editor is letting you save files with trailing whitespace. In the root of this project is an .editorconfig file. You should configure your editor to use this file and then it’ll always use the proper editor settings for whatever project you’re working on.

2) When you finish making changes locally, you can run both `bin/rails test:all` and also `bin/rubocop`. That latter one checks all the code formatting to catch issues like whitespace.

3) When you cleaned up the tests, there was one test you removed which was important to keep: a test of greater than 2 initials to ensure it only returns 2. :) I added that one back in.
